### ISSUE TITLE
"search_trigger: off" in config doesn't function; typing "off" still triggers the search

### ISSUE THREAD

**Issue by JamesMowery** [espanso/espanso#1046]
**Describe the bug**
- Edited my ~/.config/esapnso/default.yml and added search_trigger: off
- When reloading, typing "off" on the keyboard triggers the search.
- I also tried search_trigger: OFF and it still triggers when typing the same uppercase word on the keyboard

**To Reproduce**
1. Edit config and add search_trigger: off
2. Reload the config
3. Type the word "off" on the keyboard, and the search prompt comes up

**Expected behavior**
The search prompt should not pop up and be disabled, per documentation, but it continue to show up

**Setup information**
- OS: What OS are you using?
Arch Linux w/ KDE Plasma 4.24.3
- Version: which version of Espanso are you running? (you can find out by running `espanso --version` inside a terminal)
espanso 2.1.4-beta

**Comment by federico-terzi** [comment]
Hey @JamesMowery,

I'm sorry you experienced this problem! Are you using compatibility mode? That might be related. If you migrate the config the problem should go away :)

Cheers :)

**Comment by kmkrebs** [comment]
I'm having the same problem.  I have added `search_trigger: "off"` to my default.yml file, but I still get the the search window popping up.  I thought I had solved it, but I was trying to type 'coffee' and it popped up again.

OS: OS: Manjaro Linux x86_64 / Kernel: 5.10.105-1-MANJARO
Version:  espanso 2.1.5-beta

**Comment by dmfay** [comment]
Migrating the config directory structure fixed it for me (also Arch/2.1.4-beta):

```
~/.config/espanso:
├── config
│   └── default.yml
├── match
│   └── base.yml
└── user
```

If default.yml is anywhere except the `config` subdirectory it doesn't seem to process the `off` directive correctly.

There's an inconsistency in the docs, which mostly mention $CONFIG/config/default.yml as the correct location except in one of the examples where it still has the old location:

> For example, this legacy config file:
> $CONFIG/default.yml

**Comment by federico-terzi** [comment]
@dmfay Thanks! The right location for the v2 is `$CONFIG/config/default.yml` indeed, the other ones are incorrect :) You can also delete the `user` folder
